tests/library/test_artifact.py

```diff
@@ -217,3 +217,85 @@ def test_misconfigured_data_classification_policy(self):
 
         # "Fixing" the env variable so that it does not affect other tests:
         del os.environ["UNITXT_DATA_CLASSIFICATION_POLICY"]
+
+    def test_artifact_link(self):
+        with temp_catalog() as catalog_path:
+            rename = Rename(field_to_field={"old_field_name": "new_fild_name"})
+            add_to_catalog(
+                rename,
+                "rename.for.test.dict",
+                catalog_path=catalog_path,
+            )
+
+            class LinkExample1(Artifact):
+                pass
+
+            rename_fields = LinkExample1(__replacing_artifact__="rename.for.test.dict")
+            add_to_catalog(
+                rename_fields,
+                "renamefields.for.test.dict",
+                catalog_path=catalog_path,
+            )
+
+            artifact, _ = fetch_artifact("renamefields.for.test.dict")
+            self.assertDictEqual(rename.to_dict(), artifact.to_dict())
+
+    def test_artifact_link_with_deprecation_warning(self):
+        with temp_catalog() as catalog_path:
+            rename = Rename(field_to_field={"old_field_name": "new_fild_name"})
+            add_to_catalog(
+                rename,
+                "rename.for.test.dict",
+                catalog_path=catalog_path,
+            )
+
+            class LinkExample2(Artifact):
```

Comment:
There should be only one class called ArtifactLink which will be part of the official code, and be used by all users.

Reply:
Done.